### ISSUE TITLE
Modify data-validation script and dictionary-validation script to disallow exotic space characters

### ISSUE THREAD

**Issue by danpovey** [kaldi-asr/kaldi#1878]
These can cause problems later on.
This affects utils/validate_dict_dir.pl and validate_data_dir.sh.
See this forum thread:
https://groups.google.com/forum/#!topic/kaldi-help/GvYQypgWHKk

@wantee, you are good with encoding issues (although I don't know if you know perl?)
Can you help with this?

**Comment by jtrmal** [comment]
I was actually thinking about doing that, so I can work on it if no-one
else had time. Shouldn't be that hard for me.
The thing I was worried about is that for now we are fairly benevolent in
what encoding the text would be.  Implementing this too strictly might mean
that we do not allow any other encoding  than ascii and UTF8
Y.

**Comment by danpovey** [comment]
Maybe the script could first check each line to verify that there is no
character 255 or \r, and then attempt to decode as utf-8; and if this
decoding fails, assume it's OK; but if it succeeds and there are spaces
other than the "allowed" spaces (regular space and tab and newline), then
say it's OK.

BTW, I think it would be good to write any new python scripts in python3
now that we are requiring that.  Translating things about encodings later
on would be nontrivial.  Also, if you can figure out what needs to be done
and then ask @wantee, he is very good with python encodings.  But totally
your choice.

**Comment by jtrmal** [comment]
How about this: test if the document is a valid utf-8 document, which in perl can be done quite easily:
```
$foo = eval { decode("UTF-8", $foo, Encode::FB_CROAK) } ;
$valid = defined($foo); 
```
And if all the lines are valid utf08, we can test if it contains any of the individual characters which are enumerated here:
https://perldoc.perl.org/perlrecharclass.html
(in the section Whitespace).
If yes, then report error.

@wantee How do you feel about this? Do you wanna give it a shot?

**Comment by wantee** [comment]
It seems OK to me. It would be great if you can implement that, since i am
not very familiar with perl. And write a new python script may involve many
other changes in the callers.

**Comment by jtrmal** [comment]
No worries, I will look into it.

**Comment by jtrmal** [comment]
Just an info -- I have a testing version of the function that performs the aforementioned test.
Will have to test it on some databases to make sure it's not causing any harm.

**Comment by jtrmal** [comment]
being addressed in #1910 (work in progress).

**Comment by jtrmal** [comment]
resolved in #1910
